### PR TITLE
fix: [STYLE-7116] prevent setting content length header

### PR DESCRIPTION
**Pull request by ozyinc**

### CODE REVIEW

File: src/lib/utils/numericUuid.ts [hunk 1]
```diff
@@ -0,0 +1,4 @@
+function numericUuid() {
+    return Math.round(Math.random() * 10000000000000000);
```
Comment (reasn) [inline src/lib/utils/numericUuid.ts]: Let's not use `Math.random()` to create random numbers, see https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Math/random
> Math.random() does not provide cryptographically secure random numbers. Do not use them for anything related to security. Use the Web Crypto API instead, and more precisely the window.crypto.getRandomValues() method.

File: src/lib/utils/sendBeacon.ts [hunk 2]
```diff
@@ -7,26 +7,22 @@
...
-    const post = createFetch(
-        method(manifest.verb.toUpperCase()),
-        accept("application/json"),
-        json(manifest.data || {})
-    );
-
     dbg(`Sending data via "XHR".`);
-
-    await post(manifest.url);
+    await window.fetch(new Request(manifest.url), {
```
Comment (reasn) [inline src/lib/utils/sendBeacon.ts]: http-queue uses `isomorphic-fetch` to work in the browser and node.js (https://github.com/matthew-andrews/isomorphic-fetch) In the latter there's no `window`object so this would break compatibility with node.js

File: src/lib/utils/sendBeacon.ts [hunk 3]
```diff
@@ -37,7 +33,13 @@ const sendBeacon = async (manifest: IManifest): Promise<boolean> => {
 
         dbg(`Sending data via "sendBeacon" (size: ${blob.size}).`);
 
-        return nav.sendBeacon(manifest.url, blob);
+        try {
+            nav.sendBeacon(manifest.url, blob);
```
Comment (reasn) [inline src/lib/utils/sendBeacon.ts]: This again would break the library because 
> The sendBeacon() method returns true if the user agent is able to successfully queue the data for transfer, Otherwise it returns false.
> https://developer.mozilla.org/en-US/docs/Web/API/Navigator/sendBeacon

File: src/lib/utils/numericUuid.ts (hunk 1 above)
Comment (reasn) [inline src/lib/utils/numericUuid.ts]: I saw that `Math.random()` was used before. But it still is a bad idea.

Comment (reasn) [inline src/lib/utils/numericUuid.ts]: Why did you change the uuid generation?

Comment (ozyinc) [inline src/lib/utils/numericUuid.ts]: The previous one generated a string where the interface and the functions that were using it wanted a number.

File: src/lib/utils/sendBeacon.ts (hunk 2 above)
Comment (ozyinc) [inline src/lib/utils/sendBeacon.ts]: Basil.js required the window object, which is a dependancy in http-queue and it was breaking on Node so I haven't looked into the server-side consequences; but I'll change it.

File: src/lib/utils/numericUuid.ts (hunk 1 above)
Comment (reasn) [inline src/lib/utils/numericUuid.ts]: A number is okay, but then of course we must not call it uuid because it is no uuid (https://en.wikipedia.org/wiki/Universally_unique_identifier)

File: src/lib/utils/sendBeacon.spec.ts [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+ * Copyright (C) SNM Style Net Media GmbH
+ * MIT Licensed
+ */
+import {default as send} from "./sendBeacon";
+import * as sinon from "sinon";
+import IManifest from "../types/IManifest";
+declare let global: any;
+describe("sendBeacon ", () => {
+    let manifest: IManifest;
+    before(() => {
```
Comment (reasn) [inline src/lib/utils/sendBeacon.spec.ts]: I would remove `before()`, make `manifest` a `const` and assign the value directly, it's the same behavior and less code

File: src/lib/utils/sendBeacon.spec.ts [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+        global.navigator.restore();
+        global.fetch.restore();
+        global.Blob.restore();
+    });
+    it("should send request with beacon", async (done) => {
+        global.navigator.sendBeacon.resolves(true);
+        await send(manifest);
+        sinon.assert.calledOnce(global.navigator.sendBeacon);
+        sinon.assert.notCalled(global.fetch);
+        done();
```
Comment (reasn) [inline src/lib/utils/sendBeacon.spec.ts]: afaik you don't need to call `done`

File: src/lib/utils/sendBeacon.spec.ts [hunk 6]
```diff
@@ -0,0 +1,57 @@
...
+        manifest = {
+            data: {
+                mock: "data",
+            },
+            id: 1,
+            url: "/some-mock-url",
+            verb: "POST",
+        };
+    });
+    beforeEach(() => {
```
Comment (reasn) [inline src/lib/utils/sendBeacon.spec.ts]: let's always use `function` instead of `() =>` for `describe` and `it`, there is occasional weird behavior with plugins for chai/mocha so we agreed to always use `function`

File: src/lib/utils/sendBeacon.spec.ts [hunk 7]
```diff
@@ -0,0 +1,51 @@
...
+        global.fetch = sinon.stub(global, "fetch");
+    });
+    afterEach(function () {
+        global.navigator.sendBeacon.restore();
+        global.navigator.restore();
+        global.fetch.restore();
+        global.Blob.restore();
+    });
+    it("should send request with beacon", async function () {
+        global.navigator.sendBeacon.resolves(true);
```
Comment (reasn) [inline src/lib/utils/sendBeacon.spec.ts]: sendBeacon doesn't return a promise but a boolean. This stub returns a promise that is resolved with true
https://developer.mozilla.org/en-US/docs/Web/API/Navigator/sendBeacon

File: src/lib/utils/sendBeacon.spec.ts [hunk 8]
```diff
@@ -0,0 +1,51 @@
...
+        await send(manifest);
+        sinon.assert.calledOnce(global.navigator.sendBeacon);
+        sinon.assert.notCalled(global.fetch);
+    });
+    it("should send request with fetch when sendBeacon returns false", async function () {
+        global.navigator.sendBeacon.resolves(false);
```
Comment (reasn) [inline src/lib/utils/sendBeacon.spec.ts]: same here

File: src/lib/utils/sendBeacon.ts [hunk 9]
```diff
@@ -7,26 +7,22 @@
...
+        body: JSON.stringify(manifest.data || {}),
+        headers: new Headers({
+            "Accept": "application/json",
+            "Content-Type": "application/json",
+        }),
+        method: manifest.verb.toUpperCase(),
+    });
 };
 
 const sendBeacon = async (manifest: IManifest): Promise<boolean> => {
```
Comment (reasn) [inline src/lib/utils/sendBeacon.ts]: shouldn't be `async`, there's no `await`

Comment (reasn) [inline src/lib/utils/sendBeacon.ts]: (Also affects line 48 where we currently `await`
